### ISSUE TITLE
[BUG] npm ERR! enoent ENOENT: no such file or directory, lstat 'C:\Users\userName\AppData\Roaming\npm'

### ISSUE THREAD

**Issue by qll1994** [npm/cli#6971]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [x] I am using the latest npm

### Current Behavior

When "npm install" runs on pipeline server, sometimes below error occurs:
npm ERR! enoent ENOENT: no such file or directory, lstat 'C:\Users\userName\AppData\Roaming\npm'

### Expected Behavior

npm install succeeds with no error

### Steps To Reproduce

1. Newly installing node 18.17.1 with npm version: 9.6.7 which replaces node 16.17.1 and npm 8.15.0
2. Run 'npm install'
3. See error 
 npm ERR! enoent ENOENT: no such file or directory, lstat 'C:\Users\userName\AppData\Roaming\npm'

Workaround may be to create a C:\Users\userName\AppData\Roaming\npm folder manually. 

### Environment

- npm:9.6.7
- Node.js:18.17.1
- OS Name:Windows

- npm config:
```ini
; copy and paste output from `npm config ls` here
```

**Comment by wraithgar** [comment]
Not enough to reproduce here.  It looks like your node installation itself might be invalid but there's no way to know from what's here.

**Comment by mwisnicki** [comment]
@wraithgar  I can reproduce ([and we are not alone](https://stackoverflow.com/questions/25093276/node-js-windows-error-enoent-stat-c-users-rt-appdata-roaming-npm)):

```
$ npm config list
; "builtin" config from C:\Program Files\nodejs\node_modules\npm\npmrc

prefix = "C:\\Users\\mwisn\\AppData\\Roaming\\npm"

; "user" config from C:\Users\mwisn\.npmrc

cache = "C:\\Users\\mwisn\\AppData\\Local\\npm-cache"

; node bin location = C:\Program Files\nodejs\node.exe
; node version = v20.5.0
; npm local prefix = C:\Users\mwisn
; npm version = 9.8.0
; cwd = C:\Users\mwisn
; HOME = C:\Users\mwisn
; Run `npm config ls -l` to show all defaults.
```

Repro:

```
rm -r ~/AppData/Roaming/npm
npx --yes create-react-app@latest . --template typescript
```

**Comment by hamzakindi** [comment]
If you create the folder name 'npm' in C:\Users\<username>\AppData\Roaming directory. It solves the issue. My hunch is that somewhere in the script file this folder path is referred. It seems to be an orphaned entry.

**Comment by codarbind** [comment]
> If you create the folder name 'npm' in C:\Users<username>\AppData\Roaming directory. It solves the issue. My hunch is that somewhere in the script file this folder path is referred. It seems to be an orphaned entry.

this solved it for me

thanks a bunch

**Comment by SilentAssassin101** [comment]
> If you create the folder name 'npm' in C:\Users<username>\AppData\Roaming directory. It solves the issue. My hunch is that somewhere in the script file this folder path is referred. It seems to be an orphaned entry.

Thank you so much, this fixed mine as well

**Comment by SouvickMazumdar** [comment]
> If you create the folder name 'npm' in C:\Users<username>\AppData\Roaming directory. It solves the issue. My hunch is that somewhere in the script file this folder path is referred. It seems to be an orphaned entry.

Thanks a lot....It really worked

**Comment by mouadbt** [comment]
> If you create the folder name 'npm' in C:\Users<username>\AppData\Roaming directory. It solves the issue. My hunch is that somewhere in the script file this folder path is referred. It seems to be an orphaned entry.

this method worked for me

**Comment by Ifenna-Pascal** [comment]
Still having the same issues, npm already exists in the roaming folder beforehand

**Comment by faisalbhuiyan3038** [comment]
> If you create the folder name 'npm' in C:\Users<username>\AppData\Roaming directory. It solves the issue. My hunch is that somewhere in the script file this folder path is referred. It seems to be an orphaned entry.

thanks for the solution, this worked for me too

**Comment by xxshady** [comment]
~~Issue should be reopened, it's 100% reproducible on clean windows 11 right after installation of LTS nodejs~~
```
> npm -v
10.2.4
> node -v
v20.11.1
```

UPD: nvm, it fixed itself after updating npm: `npm install -g npm@latest`

**Comment by Gabmond** [comment]
> If you create the folder name 'npm' in C:\Users<username>\AppData\Roaming directory. It solves the issue. My hunch is that somewhere in the script file this folder path is referred. It seems to be an orphaned entry.

Whoa, thanks boss. I just created a new 'npm' folder and it worked.

**Comment by xxshady** [comment]
@Gabmond you could just update npm xd https://github.com/npm/cli/issues/6971#issuecomment-1976951989

**Comment by Gabmond** [comment]
> @Gabmond you could just update npm xd [#6971 (comment)](https://github.com/npm/cli/issues/6971#issuecomment-1976951989)

I already sorted it out, but thanks though.

**Comment by Michaelmayo13125152** [comment]
> If you create the folder name 'npm' in C:\Users<username>\AppData\Roaming directory. It solves the issue. My hunch is that somewhere in the script file this folder path is referred. It seems to be an orphaned entry.

this worked for me too...

**Comment by hzhang20902** [comment]
```npm install npm -g```

Plug the powerstrip into itself = infinite electricity

**Comment by hzhang20902** [comment]
> If you create the folder name 'npm' in C:\Users<username>\AppData\Roaming directory. It solves the issue. My hunch is that somewhere in the script file this folder path is referred. It seems to be an orphaned entry.

```npm install npm -g``` 
cleanest solution, now that the v9.0 bug is fixed

**Comment by Eh-haque** [comment]
> ~Issue should be reopened, it's 100% reproducible on clean windows 11 right after installation of LTS nodejs~
> 
> ```
> > npm -v
> 10.2.4
> > node -v
> v20.11.1
> ```
> 
> UPD: nvm, it fixed itself after updating npm: `npm install -g npm@latest`

this works for me.

**Comment by JessAT18** [comment]
> If you create the folder name 'npm' in C:\Users<username>\AppData\Roaming directory. It solves the issue. My hunch is that somewhere in the script file this folder path is referred. It seems to be an orphaned entry.

Thanks! This solved my problem.

**Comment by NickWoodward** [comment]
> If you create the folder name 'npm' in C:\Users<username>\AppData\Roaming directory. It solves the issue. My hunch is that somewhere in the script file this folder path is referred. It seems to be an orphaned entry.

^ worked for me, thanks!

**Comment by hzhang20902** [comment]
> If you create the folder name 'npm' in C:\Users<username>\AppData\Roaming directory. It solves the issue. My hunch is that somewhere in the script file this folder path is referred. It seems to be an orphaned entry.

`npm install npm -g` 

clean global install, the suspected root of the issue is different user priv of original node installer vs current npm locale (ie work laptop has local file system and network file system, a global install in the network file system does not propagate correctly to the local file system0

**Comment by ToufiqAmin** [comment]
> If you create the folder name 'npm' in C:\Users<username>\AppData\Roaming directory. It solves the issue. My hunch is that somewhere in the script file this folder path is referred. It seems to be an orphaned entry.

Thanks a lot! This solved my problem.

**Comment by Parkavan2003** [comment]
> If you create the folder name 'npm' in C:\Users<username>\AppData\Roaming directory. It solves the issue. My hunch is that somewhere in the script file this folder path is referred. It seems to be an orphaned entry.

Thank you so much. It solved my issue 🌟

**Comment by jericbas** [comment]
> UPD: nvm, it fixed itself after updating npm: `npm install -g npm@latest`

this is working

**Comment by ffonsecalucas** [comment]
> If you create the folder name 'npm' in C:\Users<username>\AppData\Roaming directory. It solves the issue. My hunch is that somewhere in the script file this folder path is referred. It seems to be an orphaned entry.

Thank you very much!!! It was the only thing that worked for me.

**Comment by faizullahkhan8** [comment]
> If you create the folder name 'npm' in C:\Users<username>\AppData\Roaming directory. It solves the issue. My hunch is that somewhere in the script file this folder path is referred. It seems to be an orphaned entry.

its also worked for me thanks bro

**Comment by sevilledev** [comment]
> If you create the folder name 'npm' in C:\Users<username>\AppData\Roaming directory. It solves the issue. My hunch is that somewhere in the script file this folder path is referred. It seems to be an orphaned entry.

Thanks a lot it worked for me also

**Comment by amorcayReal** [comment]
> If you create the folder name 'npm' in C:\Users<username>\AppData\Roaming directory. It solves the issue. My hunch is that somewhere in the script file this folder path is referred. It seems to be an orphaned entry.

Thanks mate it still works

**Comment by Iswaryasathish** [comment]
> If you create the folder name 'npm' in C:\Users<username>\AppData\Roaming directory. It solves the issue. My hunch is that somewhere in the script file this folder path is referred. It seems to be an orphaned entry.

It's worked thankyou

**Comment by Html5intheway4** [comment]
> If you create the folder name 'npm' in C:\Users<username>\AppData\Roaming directory. It solves the issue. My hunch is that somewhere in the script file this folder path is referred. It seems to be an orphaned entry.

Thank you, this method worked for me.🥳🎉

**Comment by VelikiiD** [comment]
> If you create the folder name 'npm' in C:\Users<username>\AppData\Roaming directory. It solves the issue. My hunch is that somewhere in the script file this folder path is referred. It seems to be an orphaned entry.

thank you, sir

**Comment by TinkerWorX** [comment]
This is still an issue, installed `Prebuilt installer v20.18.0 (LTS)` for `Windows` running `x64`. After creating the `npm` folder as suggested above, things worked as expected. Maybe the issue should be reopened @wraithgar?

**Comment by sandy0903** [comment]
> ~Issue should be reopened, it's 100% reproducible on clean windows 11 right after installation of LTS nodejs~
> 
> ```
> > npm -v
> 10.2.4
> > node -v
> v20.11.1
> ```
> 
> UPD: nvm, it fixed itself after updating npm: `npm install -g npm@latest`

it worked for me thanks alot

**Comment by samuelrayo** [comment]
> If you create the folder name 'npm' in C:\Users<username>\AppData\Roaming directory. It solves the issue. My hunch is that somewhere in the script file this folder path is referred. It seems to be an orphaned entry.

Thank you sir!

**Comment by eabase** [comment]
@wraithgar 

Come on guys! 
Why was this issue closed if it was never fixed?

I use powershell core (`pwsh.exe`) for everything on Win-11, and ran into the same issue just now. 
I installed the MSI version as normal user (and not in `C:\Program Files\`). 

Why doesn't the installer create this directory?
It managed to create the other one (`C:\Users\xxxx\AppData\Local\npm-cache`) fine, so 
I don't see why it can't create  `C:\Users\xxxx\AppData\Roaming\npm` as well. 

Possibly related issue: 

```powershell
# where.exe npm
C:\lang\nodejs\npm       # Bash script (not used in pwsh!)
C:\lang\nodejs\npm.cmd   # CMD script (deprecated on Windows, but THIS is used!)
```

In other words, running `npm` in powershell actually runs the old CMD (`npm.bat`) version. 
But **WHY?** You already have a `npm.ps1` in the same directory...
